specs/merge/fork-choice.md

```diff
@@ -119,11 +135,12 @@ def on_block(store: Store, signed_block: SignedBeaconBlock) -> None:
     assert get_ancestor(store, block.parent_root, finalized_slot) == store.finalized_checkpoint.root
     
     # [New in Merge]
-    if is_transition_block(pre_state, block):
+    is_transition_store_initialized = transition_store is not None
+    if is_transition_store_initialized and is_transition_block(pre_state, block):
         # Delay consideration of block until PoW block is processed by the PoW node
         pow_block = get_pow_block(block.body.execution_payload.parent_hash)
         assert pow_block.is_processed
-        assert is_valid_transition_block(pow_block)
+        assert is_valid_transition_block(transition_store, pow_block)
```

Comment:
Our type checker (`mypy`) is kinda silly here...
If you change it to:
```python
    if transition_store is not None and is_transition_block(pre_state, block):
        # Delay consideration of block until PoW block is processed by the PoW node
        pow_block = get_pow_block(block.body.execution_payload.parent_hash)
        assert pow_block.is_processed
        assert is_valid_transition_block(transition_store, pow_block)
```
then the `mypy` will pass. 😅